memphis/memphis.py

```diff
@@ -180,14 +187,42 @@ async def producer(self, station_name, producer_name, generate_random_suffix=Fal
                 createProducerReq, indent=2).encode('utf-8')
             err_msg = await self.broker_manager.request("$memphis_producer_creations", create_producer_req_bytes)
             err_msg = err_msg.data.decode("utf-8")
-
-            if err_msg != "":
+            err_msg = json.loads(err_msg)
+            if err_msg['error'] != "":
                 raise Exception(err_msg)
+            
+            await self.start_listen_for_schema_updates(station_name)
             return Producer(self, producer_name, station_name)
 
         except Exception as e:
             raise Exception(e)
 
+    async def get_msg_schema_updates(self, station_name_internal, iterable):
+        self.schema_updates_data[station_name_internal] =  {}
+        async for msg in iterable:
+            message = msg.data.decode("utf-8")
+            message = json.loads(message)
+            empty = message['UpdateType'] == 2
+            if empty == 2:
+                data = {}
+            else:
+                data = message['init']
+            self.schema_updates_data[station_name_internal] =  data
+
+    async def start_listen_for_schema_updates(self, station_name):
+        station_name_internal = get_internal_name(station_name)
+        schema_updates_subject = "$memphis_schema_updates_"+station_name_internal
+
+        schema_exists = self.schema_updates_subs.get(station_name)
```

Comment:
in some places your keys are the externally station name, please change it to internally always